content/20.zksync-protocol/20.contracts/31.evm-emulator/00.overview.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Overview
+description: General details of the EVM Bytecode Emulator
+---
+## EVM Bytecode Emulator
+
+ZK chains like Era operate on **EraVM**, a ZK-optimized virtual machine that differs from the Ethereum Virtual Machine (EVM)
+in its instruction set and execution model. While Solidity and Vyper can be compiled to EraVM bytecode, differences in
+execution and tooling have required modifications in some cases.
+
+To address this, ZKsync introduces **EVM execution mode via an EVM Bytecode Emulator**, allowing **unmodified EVM bytecode** to run
+on ZK chains **without recompilation or changes to developer tooling**. This enables projects built for Ethereum to execute in
+ZKsync while maintaining EraVM as the core execution environment.
+
+## Execution Model
+
+- **EraVM remains the primary execution environment.** The EVM Emulator does not replace EraVM but acts as a **translation layer**,
+  enabling the execution of EVM bytecode on top of EraVM.
+- **EVM contracts are flagged for emulation.** When an EVM contract is deployed, its bytecode hash is marked with a special identifier.
+  Instead of running native EraVM bytecode, the system routes execution through the emulator.
+- **EVM opcodes are interpreted at runtime.** The emulator translates EVM instructions into **EraVM operations**, ensuring execution
+  adheres to EVM semantics as closely as possible.
+- **Gas accounting is based on EraVM (ergs).** While transaction fees remain lower than Layer 1, the cost of executing EVM contracts
```

Comment:
Everyting related to gas is confusing (because our gas model is confusing). We need to think about it a bit. Maybe we should have a separate page with a clear explanation of how our gas model works